src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -117,10 +117,10 @@ JvmtiThreadState::JvmtiThreadState(JavaThread* thread, oop thread_oop)
     _is_virtual = java_lang_VirtualThread::is_instance(thread_oop);
   }
 
-  // thread can be NULL if virtual thread is unmounted
   if (thread != NULL) {
-    // set this as the state for the thread only if thread_oop is current thread->jvmti_vthread()
     if (thread_oop == NULL || thread->jvmti_vthread() == NULL || thread->jvmti_vthread() == thread_oop) {
+      // Set this as the state for the JavaThread or mounted virtual thread
+      // only if thread_oop is current thread->jvmti_vthread().
       thread->set_jvmti_thread_state(this);
```

Comment:
This update has the same problem as previous one that I've commented. We set the thread->jvmti_thread_state() value in the JavaThread when it is bound to a normal/carrier or mounted virtual thread. There is no JavaThread only for unmounted virtual thread.

Reply:
I've updated the comment to:
```
      // The JavaThread for carrier or mounted virtual thread case.
      // Set this only if thread_oop is current thread->jvmti_vthread().
```
to try and match your style from your previous comment. Please let me know
if the new version works for you.